test/test_profiler.py

```diff
@@ -99,6 +101,62 @@ def forward(self, x):
 
         torch._C._set_graph_executor_optimize(prev_opt)
 
```

Comment:
add test for custom optimizer

Reply:
I prefer less code, because less code will bring less bug. Let's add it if reviewer ask for, OK?